### ISSUE TITLE
Use 2-3 times more memory than Spectacle

### ISSUE THREAD

**Issue by mickaphd** [rxhanson/Rectangle#4]
Works fine but I don't see any significant differences with Spectacle in terms of reactivity. Also the app uses more memory than Spectacle (not really a problem but still).

**Comment by rxhanson** [comment]
Can you explain what you mean by reactivity? Is it time between shortcut execution and window movement?

The increased memory that you see in Rectangle is mostly due to a bug where the memory used by the preferences window is not released when the window is closed. A fix for this can go in the next release. 

You'll see a reduced memory footprint if the app is run without opening the preferences window. Beyond that, the memory usage should be pretty close to Spectacle and differences should mainly be with respect to Swift vs Obj-C.

**Comment by mickaphd** [comment]
Indeed, but I have to admit after a few more hours of test that Rectangle seems maybe faster than Spectacle..!
Great for the memory, promising.

**Comment by rxhanson** [comment]
Ok, I finally got a chance to dig into the memory management of Rectangle, and I did some profiling and comparison to Spectacle. 

I performed the profiling of both apps from the Instrumentation available in Xcode, and for my simple testing Rectangle actually operated with less memory footprint than Spectacle (although very slightly). On modern systems, we're really splitting hairs here to start with - and the differences pretty negligible, but here's what I found. 

Without a prefs window open, Rectangle ran with about 14MB, and Spectacle ran with about 14.7MB. I noticed someone on [macg](https://www.macg.co/logiciels/2019/07/rectangle-un-nouveau-gestionnaire-de-fenetres-gratuit-107105) saying that Spectacle ran with 8MB, and I'm not sure why I would see a different number, but that's a pretty small difference for most people anyway. 

When a prefs window opens, Rectangle goes up to around 25-26MB, and Spectacle goes up very slightly above that. As I mentioned, I had a bug in Rectangle where the memory wasn't released when the window was closed. Interestingly enough, Spectacle had the same bug!

I've modified Rectangle to release the memory for the prefs window, and the Accessibility window that comes up on first launch. However, when these windows are deallocated, the memory usage doesn't go back down to the 14MB, it only goes down to about 22MB. From what I can tell, this is just that macOS isn't deallocating everything. My understanding is that this is by design.

In digging through what's actually using memory in Rectangle, there's really not much optimization that I feel like needs to be done. The only things in memory that could be removed and loaded as needed are the window movers and the window frame calculations. They are really small though (negligible), and it makes sense to me to leave them loaded in memory because that's going to keep the speed of the window movement up.

On window movement speed, it's no surprise to me that you would see faster times than Spectacle. Spectacle loaded a JavaScript engine in memory and would use JavaScript to calculate the window size/position. I have no idea why that would be done in Spectacle, but that's definitely not going to be faster than Rectangle, which does all of these calculations in Swift. It might be unnoticeable on some machines, but noticeable on others.

I checked in the change to release the memory from the windows, but I haven't put out a new release yet - I have a few more changes to add before it's worth a release. I'm looking to close out this issue as long as you're ok with it.
